doc/source/data/package-ref.rst

```diff
@@ -1,6 +1,6 @@
 .. _data_api:
 
-Ray Data API
+Ray Datasets API
```

Comment:
Same here. This should(!) be the Ray Data API, i.e. the API for the whole thing. Just because we don't currently have API refs for the other parts of Ray Data (which I'm still not sure what that's supposed to be, other than integrations), doesn't mean we have to give up on consistency.

Top level: Ray Data API

Sub-section: Ray Datasets API

Again, otherwise how would users have a chance of understanding this?